xdsl/ir/core.py

```diff
@@ -407,6 +392,20 @@ def __str__(self) -> str:
         return res.getvalue()
 
 
+class TypeAttribute(Attribute):
+    """
+    This class should only be inherited by classes inheriting Attribute.
```

Comment:
How is this enforced now? It used to be a `__post_init__` check.

Reply:
I think just by subclassing Attribute

Reply:
This is not really enforced anymore after this change I believe :/
The main thing is that `Attribute` classes should either be `Data` or `ParametrizedAttribute`, and we should probably just enforce that in general.
I'm fine removing the check here, and having a check somewhere else at some point.

Reply:
Maybe track this with a gh issue?

Reply:
Right @math-fehr, you mentionned that already and it slipped through, my bad.
I think I would prefer the same check but on Attribute himself then? I'll propose something

Reply:
Oh but wait; how was this check enforcing this?
All it did was checking that it was an `Attribute` instance, not specifically a `Data` or `ParameterizedAttribute`.
But, same answer; I'll propose something to check this specifically, at the `Attribute` level.

Reply:
Okay, this is already enforced at the IRDL level; just adding a test for it

Reply:
Do we have non-IRDL attributes like we have non-IRDL Operations?

Reply:
Pushed a simple test for the existing IRDL enforcment. Not sure if it would make more sense at the Attribute level?